sdk/aqueduct/integrations/sql_integration.py

```diff
@@ -127,14 +145,43 @@ def sql(
                 query=extract_params,
             )
 
-        operator_id = generate_uuid()
-        output_artifact_id = generate_uuid()
+        # Find any tags that need to be expanded in the query, and add the parameters that correspond
+        # to these tags as inputs to this operator. The orchestration engine will perform the replacement at runtime.
+        sql_input_artifact_ids = []
+        if extract_params.query is not None:
+            matches = re.findall(TAG_PATTERN, extract_params.query)
+            for match in matches:
+                param_name = match.strip(" {}")
+
+                # If it is a built-in tag, we can ignore it for now, since the python operators will perform the expansion.
+                if param_name in BUILT_IN_EXPANSIONS:
+                    continue
+
+                param_op = self._dag.get_operator(with_name=param_name)
+                if param_op is None:
+                    raise InvalidUserArgumentException(
+                        "There is no parameter defined with name `%s`." % param_name,
+                    )
+
+                # Check that the parameter corresponds to a string value.
+                assert param_op.spec.param is not None
+                param_val = json.loads(param_op.spec.param.val)
+                if not isinstance(param_val, str):
+                    raise InvalidUserArgumentException(
+                        "The parameter `%s` must be defined as a string. Instead, got type %s"
```

Comment:
Is this checking applied to the default value? who would perform the similar checking in `run()` ?

Reply:
The `UpdateParametersDelta` will perform this same check. Thats the delta that changes the parameter value temporarily, and it performs the string check if the parameter is consumed by any extract operator.